operator/cilium_node.go

```diff
@@ -468,3 +489,39 @@ func updateCNP(ciliumClient v2.CiliumV2Interface, cnp *cilium_v2.CiliumNetworkPo
 		}
 	}
 }
+
+func ClearCnpStatusNodes(ctx context.Context) error {
+	clearStatusNodes := func(ctx context.Context, cnp *cilium_v2.CiliumNetworkPolicy) error {
+		data, err := json.Marshal([]k8s.JSONPatch{
+			{
+				OP:   "remove",
+				Path: "/status/nodes",
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// If the namespace is empty the policy is the clusterwide policy
+		// and not the namespaced CiliumNetworkPolicy.
+		if namespace := utils.ExtractNamespace(&cnp.ObjectMeta); namespace == "" {
+			_, err = ciliumK8sClient.CiliumV2().CiliumClusterwideNetworkPolicies().Patch(ctx,
+				cnp.Name, types.JSONPatchType, data, meta_v1.PatchOptions{}, "status")
+		} else {
+			_, err = ciliumK8sClient.CiliumV2().CiliumNetworkPolicies(namespace).Patch(ctx,
+				cnp.Name, types.JSONPatchType, data, meta_v1.PatchOptions{}, "status")
+		}
+
+		return err
+	}
+
+	cnpsAtOnce := 10
+	if err := forEachCNP(ctx, true, cnpsAtOnce, clearStatusNodes); err != nil {
+		return err
+	}
+	if err := forEachCNP(ctx, false, cnpsAtOnce, clearStatusNodes); err != nil {
+		return err
```

Comment:
This is a best-effort operator, whatever errors we get we shouldn't do an early return. Also, there are certain errors that we can ignore, like `k8s.io/apimachinery/pkg/api/errors.IsNotFound` for example.